gateway/src/main/java/io/camunda/zeebe/gateway/ResponseMapper.java

```diff
@@ -276,22 +278,30 @@ public static SetVariablesResponse toSetVariablesResponse(
     return SetVariablesResponse.newBuilder().setKey(key).build();
   }
 
-  public static ActivateJobsResponse toActivateJobsResponse(
-      final long key, final JobBatchRecord brokerResponse) {
+  public static JobActivationResult toActivateJobsResponse(
+      final long key, final JobBatchRecord brokerResponse, final int maxMessageSize) {
     final ActivateJobsResponse.Builder responseBuilder = ActivateJobsResponse.newBuilder();
 
     final Iterator<LongValue> jobKeys = brokerResponse.jobKeys().iterator();
     final Iterator<JobRecord> jobs = brokerResponse.jobs().iterator();
 
+    int currentResponseSize = 0;
```

Comment:
❌ I believe this is not entirely correct. I assume there is some metadata associated to an `ActivateJobsResponse`. Currently, only the size of the individual `ActivatedJob`s are taken into account.

You could try creating an empty `ActivateJobsResponse` object to see the serialized size, and then doing the same with one, two, and three added `ActivatedJob`s to see how it grows. This will give you the insight what the initial size should be.

Reply:
@korthout 


```java
final ActivateJobsResponse response =
        ActivateJobsResponse.newBuilder()
            .addJobs(ActivatedJob.newBuilder().build())
            .build();
System.out.println("response size: " + response.getSerializedSize()); // prints 2

final ActivatedJob activatedJob1 = ActivatedJob.newBuilder().build();
    System.out.println("activatedJob empty size: " + activatedJob1.getSerializedSize()); // prints 0
```

It only adds 2 bytes for each job added to ActivateJobsResponse and 0 for an empty activated job. I updated the code based on that, please take a look. 👀